### PR TITLE
Add new ModalBlock

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Add new ModalBlock to be able to wrap other blocks in a Modal
Will talk with William and Peter to decide what blocks we want to be able to use in the modal

https://github.com/HedvigInsurance/racoon/assets/6661511/e60a83df-22df-449b-824f-defd815dc23a

## Justify why they are needed
Enable SelectInsuranceGridBlock on startpage 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2333** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2333" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2333?utm_source=stack-comment).

File: apps/store/src/blocks/ModalBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      <FullscreenDialog.Root open={isOpen} onOpenChange={setIsOpen}>
+        <FullscreenDialog.Trigger asChild>
+          <Button variant={blok.buttonVariant ?? 'primary'} size={blok.buttonSize ?? 'medium'}>
+            {blok.buttonText}
+          </Button>
+        </FullscreenDialog.Trigger>
+
+        <FullscreenDialog.Modal center={true}>
+          {blok.modalContent.map((nestedBlock) => (
+            <StoryblokComponent blok={nestedBlock} key={nestedBlock._uid} />
```
Comment (alebedev) [inline apps/store/src/blocks/ModalBlock.tsx:29]: Nit: let's keep `key` first, easier to read

**Review by alebedev — APPROVED** [review]
Nice!

**Comment by gustaveen** [comment]
@gustaveen merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2333).

**Pull request merged**